pygments/lexers/scripting.py

```diff
@@ -57,7 +58,13 @@ class LuaLexer(RegexLexer):
 
     _comment_multiline = r'(?:--\[(?P<level>=*)\[[\w\W]*?\](?P=level)\])'
     _comment_single = r'(?:--.*$)'
-    _space = r'(?:\s+)'
+    if sys.version_info >= (3, 11):
+        # Use a possessive quantifier to prevent greediness from causing runaway backtracking.
+        _space = r'(?:\s++)'
+    else:
+        # Possessive quantifiers are not available in Python < 3.11. Complex Lua is likely to
+        # cause any regex that includes _space to take an exponentially long time.
+        _space = r'(?:\s+)'
```

Comment:
A negative lookahead assertion works in all Python versions.

```python
    _space = r'(?:\s+(?!\s))'
```

Reply:
Nice that makes sense. I fixed it up and it works as far as I can tell. Thanks for helping simplify.

Reply:
Tested, works here - appreciate the investigation!